pkg/credentialprovider/plugin/config_test.go

```diff
@@ -312,7 +312,27 @@ providers:
     - name: FOO
       value: BAR`,
 			config:    nil,
-			expectErr: true,
+			expectErr: `no kind "CredentialProviderConfig" is registered for version "foobar/v1alpha1`,
+		},
+		{
+			name: "config with invalid typo",
+			configData: `---
+kind: CredentialProviderConfig
+apiVersion: kubelet.config.k8s.io/v1
+providers:
+  - name: test
```

Comment:
We also want to detect duplicate names right?

Reply:
that's being handled today while registering the providers (after reading and validating the config)

https://github.com/kubernetes/kubernetes/blob/78879fa3d088def7a0ad51d13b42a20a6d46def9/pkg/credentialprovider/plugins.go#L40-L43

but we could move the duplicate name check to validation of config, so we return early?

Reply:
Ah okay, we can leave it as-is for now.